frontend/src/metabase/common/hooks/use-homepage-dashboard.tsx

```diff
@@ -0,0 +1,26 @@
+import { useLocation } from "react-use";
+
+import { skipToken, useGetDashboardQuery } from "metabase/api";
+import { useSelector } from "metabase/lib/redux";
+import { getCustomHomePageDashboardId } from "metabase/selectors/app";
+import { getSettingsLoading } from "metabase/selectors/settings";
+
+export const useHomepageDashboard = () => {
+  const dashboardId = useSelector(getCustomHomePageDashboardId);
+  const isLoadingSettings = useSelector(getSettingsLoading);
+  const { pathname } = useLocation();
+
+  const { data: dashboard, isLoading: isLoadingDashboard } =
+    useGetDashboardQuery(dashboardId ? { id: dashboardId } : skipToken);
+
+  const isAtHomepageDashboard = Boolean(
+    dashboardId && pathname?.startsWith(`/dashboard/${dashboardId}`),
```

Comment:
I assume `.startsWith(`/dashboard/${dashboardId}`)` may not work correctly when a metabase instance is not in under the root path, for instance `https://myapp.com/analytics/`